scylla-cdc/src/customer.rs

```diff
@@ -0,0 +1,113 @@
+        //          and therefore we can't map the result
+        //          to something other than a hardcoded tuple.
+
+        // [TODO]:  We might want to extract this so that the mapping
+        //          is created once per query, not once per row.
+        let mut mapping: HashMap<String, usize> = HashMap::new();
+        let mut i: usize = 0;
+        let mut data: Vec<CqlValue> = Vec::new();
+
+        // If cdc read was successful, these default values will not be used.
+        // [TODO]: Test it.
+        let mut stream_id: Vec<u8> = vec![];
+        let mut time: uuid::Uuid = uuid::Uuid::default();
+        let mut batch_seq_no: i32 = 0;
+        let mut end_of_batch: bool = false;
+        let mut operation: OperationType = OperationType::PreImage;
+        let mut ttl: Option<i64> = None;
+
+        for spec in specs.into_iter() {
+            let curr = row.columns[i].as_ref().unwrap();
+            match spec.name.as_str() {  // spec.name is public since 0.3.0 driver version, it didn't work on 0.2.1.
+                STREAM_ID_NAME => stream_id = curr.as_blob().unwrap().to_vec(),
+                TIME_NAME => time = curr.as_uuid().unwrap(),
+                BATCH_SEQ_NO_NAME => batch_seq_no = curr.as_int().unwrap(),
+                END_OF_BATCH_NAME => end_of_batch = curr.as_boolean().unwrap(),
+                // [FIXME]: This way of creating an enum does not work, write another function.
+                OPERATION_NAME => operation = OperationType(curr.as_tinyint().unwrap()),
+                TTL_NAME => ttl = curr.as_bigint(), // as_bigint returns Some(i) if the type applies, and None if not.
+                x @ _ => {
+                    mapping.insert(String::from(x), i);
+                    // [FIXME]  CqlValue does not implement or derive Copy, so we can't do this.
+                    data.push(curr.clone());
+                    i += 1;
+                },
+            }
+        }
+
+        CDCRow {
+            stream_id,
+            time,
+            batch_seq_no,
+            end_of_batch,
+            operation,
+            ttl,
+            data,
+            mapping,
+        }
+    }
+
+    // Allows to get a value from the column that corresponds to the logged table.
+    pub fn get_value<T>(&self, name: &str) -> Option<Option<T>> {
+        // [TODO]: Create a way to return a generic value, maybe use an inspiration from Driver's code.
+        let index = self.mapping.get(name);
+        return match index {
+            Some(id) => Some(self.data[id.clone()]),
+            None => None,
+        }
+    }
+
+    // Allows to get info if a value was deleted in this operation.
+    pub fn is_value_deleted(&self, name: &str) -> Option<Option<bool>> {
+        let index = self.mapping.get(&(IS_DELETED_PREFIX.to_owned() + name));
+        return match index {
+            Some(id) => Some(self.data[id.clone()].as_boolean()),
+            None => None,
+        }
+    }
```

Comment:
We need another getter function, for the `cdc$deleted_elements_xyz` function. See the "Advanced column types" section of the CDC documentation for more info: https://docs.scylladb.com/using-scylla/cdc/cdc-advanced-types/#advanced-column-types.